### PR TITLE
Properly enable/disable metrics and health endpoints

### LINKED ISSUE #11506: Properly enable/disable metrics and health endpoints
### Description

The health and metrics endpoints are always available even though the options `health-enabled` and `metrics-enabled` are set to `false`, respectively.

This is because Quarkus exposes these endpoints whenever the extensions (health and metrics) are installed to applications.

The options we are using from Quarkus to disable/disable metrics and health ([quarkus.smallrye-metrics.extensions.enabled](https://quarkus.io/guides/smallrye-metrics#quarkus-smallrye-metrics_quarkus.smallrye-metrics.extensions.enabled) and [quarkus.health.extensions.enabled](https://quarkus.io/guides/smallrye-health#quarkus-smallrye-health_quarkus.health.extensions.enabled)) only control if items should be loaded from extensions but not disable the capability completly.

### Discussion

_No response_

### Motivation

For now, keep the `health-enabled` and `metrics-enabled` options aligned with their purpose in order to completely disable these endpoints if they are set to `false`.

### Details

There are other open points for health that should be addressed by a follow-up epic/issues. Right now, only fixing the current behavior.

For instance, I'm not even sure if it makes sense to disable health and metrics considering how they are used in production. Even for dev, we should make them available, IMO.

### PR DESCRIPTION
**Pull request by pedroigor**
Closes #11506

* Completly remove the health and metrics endpoints when they are disabled
* Fixes the `health/live` and `health/ready` endpoints to only show their specific checks
* Avoid executing health and metrics endpoint in a worker thread as they run in their own worker thread, and avoid creating a `KeycloakSession` every time those endpoints are invoked. The current behavior is causing transactions to not disassociate from the calling thread and causing Arjuna to wait for them to timeout. Eventually, the server might block during the shutdown.
* Fixes the `KeycloakSecurityHeadersFilter` in case the health and metrics endpoints are invoked where there is no session from the calling thread
* Update docs with the current status of our health endpoint

There are many improvements to health and metrics we can do. This PR basically brings some consistency to what we have and fixes some known issues. 

For health, we should be improving it as per https://github.com/keycloak/keycloak/discussions/11556.

### CODE REVIEW

**Review by DGuhr — CHANGES_REQUESTED** [review]
Thanks! A few things I wanted to ask about this one to understand the changes better, a typofix, but in general LGTM :)

File: quarkus/deployment/src/main/java/org/keycloak/quarkus/deployment/KeycloakProcessor.java [hunk 1]
```diff
@@ -482,48 +476,36 @@ void indexJpaStore(BuildProducer<IndexDependencyBuildItem> indexDependencyBuildI
         indexDependencyBuildItemBuildProducer.produce(new IndexDependencyBuildItem("org.keycloak", "keycloak-model-map-jpa"));
     }
 
+    @Record(ExecutionTime.STATIC_INIT)
     @BuildStep
-    void initializeFilter(BuildProducer<FilterBuildItem> filters, LaunchModeBuildItem launchModeBuildItem) {
-        QuarkusRequestFilter filter = new QuarkusRequestFilter();
-        LaunchMode launchMode = launchModeBuildItem.getLaunchMode();
+    void initializeFilter(BuildProducer<FilterBuildItem> filters, KeycloakRecorder recorder) {
+        filters.produce(new FilterBuildItem(recorder.createRequestFilter(isHealthEnabled() || isMetricsEnabled()),FilterBuildItem.AUTHORIZATION - 10));
```
[the remark this answers is not in the document]
Comment (pedroigor) [reply quarkus/deployment/src/main/java/org/keycloak/quarkus/deployment/KeycloakProcessor.java:482]: It is always called before "functional" routes. The order there is for routes behaving as filters, basically non-functional routes dealing with some cross-cutting concerns like authentication, cord, and authz.

File: quarkus/runtime/src/main/resources/application.properties [hunk 2]
```diff
@@ -24,3 +28,6 @@ quarkus.package.include-dependency-list=false
 
 # we do not want running dev services in distribution
 quarkus.devservices.enabled=false
+
+# We want to expose non-application paths (e.g. health) at the root path
+quarkus.http.non-application-root-path=${quarkus.http.root-path}
```
[the remark this answers is not in the document]
Comment (pedroigor) [reply quarkus/runtime/src/main/resources/application.properties:33]: Do we need to disable it at all?

File: quarkus/tests/integration/src/main/java/org/keycloak/it/utils/RawKeycloakDistribution.java [hunk 3]
```diff
@@ -320,8 +320,8 @@ private void reset() {
         outputStream.clear();
         errorStream.clear();
         exitCode = -1;
-        keycloak = null;
         shutdownOutputExecutor();
+        keycloak = null;
```
[the remark this answers is not in the document]
Comment (pedroigor) [reply quarkus/tests/integration/src/main/java/org/keycloak/it/utils/RawKeycloakDistribution.java:324]: Concurrency, async read can happen after keycloak is set to null.

**Review by DGuhr — DISMISSED** [review]
LGTM, thanks :)

**Review by abstractj — DISMISSED** [review]

File: quarkus/deployment/src/main/java/org/keycloak/quarkus/deployment/KeycloakProcessor.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (DGuhr) [reply quarkus/deployment/src/main/java/org/keycloak/quarkus/deployment/KeycloakProcessor.java:482]: thx, so basically same concept :)

**Review by DGuhr — APPROVED** [review]
LGTM

**Pull request merged**